Observer/BeforeSending.php

```diff
@@ -30,16 +30,18 @@ public function execute(Observer $observer): void
         $event = $observer->getEvent()->getSentryEvent()->getEvent();
         $hint = $observer->getEvent()->getSentryEvent()->getHint();
 
-        $hintMessage = $hint->exception->getMessage();
-        if ($hint->exception instanceof LocalizedException) {
-            $hintMessage = $hint->exception->getRawMessage();
-            $event->setMessage($hintMessage);
-        }
+        if ($hint?->exception !== null) {
+            $hintMessage = $hint->exception->getMessage();
+            if ($hint->exception instanceof LocalizedException) {
+                $hintMessage = $hint->exception->getRawMessage();
+                $event->setMessage($hintMessage);
+            }
```

Comment:
Aren't we able to get the message some other way when we do not have the exception, like getting it from the Event itself?
Then at least we will be able to continue to filter out events even without hint.
```suggestion
        $hintMessage = $hint?->exception?->getMessage() ?? $event->getMessage();
        if ($hint?->exception instanceof LocalizedException) {
            $hintMessage = $hint->exception->getRawMessage() ;
            $event->setMessage($hintMessage);
        }
```

Otherwise i would suggest flipping the logic of the if and returning early to reduct the depth of if/else
```suggestion
        if ($hint?->exception === null) {
            return
        }
        $hintMessage = $hint->exception->getMessage();
        if ($hint->exception instanceof LocalizedException) {
            $hintMessage = $hint->exception->getRawMessage();
            $event->setMessage($hintMessage);
        }
```